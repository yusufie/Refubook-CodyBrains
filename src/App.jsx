import React from 'react';
import { Routes, Route } from 'react-router-dom';
import {
  About,
  Auth,
  Blog,
  Contact,
  Home,
  NotFound,
  SingleBlogPage,
  UserProfile,
  Write,
} from './pages';
import { Footer, Navbar } from './components';

function App() {
  return (
    <div className="w-full  min-h-screen flex flex-col gap-2 ">
      <Navbar />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="about" element={<About />} />
        <Route path="blog" element={<Blog />} />
        <Route path="contact" element={<Contact />} />
        {['signin', 'signup'].map((path) => (
          <Route
            key={path}
            path={path}
            element={<Auth isLogin={path === 'signin'} />}
          />
        ))}
        <Route path="profile" element={<UserProfile />} />
        <Route path="*" element={<NotFound />} />
        <Route path="write" element={<Write />} />
        <Route path="blog/:id" element={<SingleBlogPage />} />
      </Routes>
      <Footer />
    </div>
  );
}

export default App;
