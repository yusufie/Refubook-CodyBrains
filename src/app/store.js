import { configureStore } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import storage from 'redux-persist/lib/storage';
import userReducer from '../features/userSlice';

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, userReducer);

const store = configureStore({
  reducer: {
    user: persistedReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});

export const persistor = persistStore(store);

setupListeners(store.dispatch);

export default store;
