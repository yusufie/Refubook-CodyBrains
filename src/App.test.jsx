import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import App from './App';
import store from './app/store';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Routes: jest.fn(({ children }) => <div>{children}</div>),
  Route: jest.fn(({ children }) => <div>{children}</div>),
  Link: jest.fn(({ children }) => <a href="/testpath">{children}</a>),

  useNavigate: () => jest.fn(),
  useLocation: () => ({
    pathname: '/',
  }),
}));

jest.mock('react-i18next', () => ({
  ...jest.requireActual('react-i18next'),
  useTranslation: () => ({
    t: (key) => key,
    i18n: {
      changeLanguage: () => new Promise(() => {}),
    },
  }),
}));

test('renders App', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
