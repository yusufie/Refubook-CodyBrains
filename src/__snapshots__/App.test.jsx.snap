// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<DocumentFragment>
  <div
    class="w-full  min-h-screen flex flex-col gap-2 "
  >
    <div
      class=" w-full  z-[100] py-6 px-8"
    >
      <div
        class="flex justify-between  items-center w-full h-full "
      >
        <div
          class="flex items-center"
        >
          <button
            class="md:hidden block cursor-pointer  "
            type="button"
          >
            <svg
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              />
            </svg>
          </button>
        </div>
        <div
          class="hidden md:flex items-center gap-4"
        >
          <ul
            class="hidden md:flex"
          />
          <div
            class="flex items-center gap-4"
          />
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="fixed left-[-100%] top-0 bottom-0 p-10 ease-in duration-500 bg-refubookWhite"
        >
          <div>
            <div
              class="flex w-full items-center "
            >
              <button
                class="rounded-full  p-3 cursor-pointer text-refubookBlue text-2xl"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="py-4 flex flex-col h-full"
          >
            <div
              class="flex items-center justify-between w-full"
            />
            <ul
              class="flex flex-col ml-5 mt-28 h-full "
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col  gap-5 px-14 py-6 mb-6 lg:flex-row  w-full bg-refubookWhite mt-10"
    >
      <div
        class="items-center justify-center flex mr-6"
      />
      <ul
        class="flex justify-center gap-5 items-center text-md md:text-lg xl:text-xl 2xl:text-2xl  w-full"
      >
        <li />
        <li />
        <li />
        <li />
      </ul>
      <div
        class="flex w-full justify-center gap-5 md:gap-10  lg:justify-end"
      >
        <div
          class="flex items-center justify-center relative"
        >
          <button
            aria-expanded="false"
            aria-haspopup="listbox"
            class="  flex items-center justify-between gap-2 font-medium z-40  bg-refubookBlue text-refubookWhite rounded-full py-2  md:px-10 px-3 cursor-pointer  duration-300 ease-in"
            data-headlessui-state=""
            id="headlessui-listbox-button-:r0:"
            type="button"
          >
            <span
              class="text-sm md:text-lg "
            >
               English
            </span>
            <span
              class="flag-icon flag-icon-gb mx-2"
            />
            <svg
              class="w-5 h-5"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.646 4.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1-.708.708L8 5.707l-5.646 5.647a.5.5 0 0 1-.708-.708l6-6z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
